coffee_cmd/src/main.rs

```diff
@@ -27,14 +27,20 @@ async fn run(args: CoffeeArgs, mut coffee: CoffeeManager) -> Result<(), CoffeeEr
                 None
             };
             let result = coffee.install(&plugin, verbose, dynamic).await;
```

Comment:
```suggestion
            let result = coffee.install(&plugin, verbose, dynamic).await?;
```

I think just this is enough? to get the error back?

Reply:
Well yes, but then the code after needs changes anyways:
```
coffee.install(&plugin, verbose, dynamic).await?;
if let Some(spinner) = spinner {
    spinner.finish();
} else {
    term::success!("Plugin {plugin} Compiled and Installed")
}
```
I can force push this version if you like it more.

Reply:
yes I like your diff except if we can do `spinner.and_then(|spinner| spinnet.finish());`

Reply:
which diff do you like 1. or 2. ? Also no clue why and where you want that line. It would need to be 
```rust
spinner.and_then(|spin| Some(spin.finish()));
```
and even then clippy is yelling at me.

Reply:
>and even then clippy is yelling at me.

No I do not think so we do something similar inside the upgrade command, IMHO the install code should be the same.

This is a diff that works

```diff 
diff --git a/coffee_cmd/src/main.rs b/coffee_cmd/src/main.rs
index d5708fa..238e897 100644
--- a/coffee_cmd/src/main.rs
+++ b/coffee_cmd/src/main.rs
@@ -26,15 +26,15 @@ async fn run(args: CoffeeArgs, mut coffee: CoffeeManager) -> Result<(), CoffeeEr
             } else {
                 None
             };
-            let result = coffee.install(&plugin, verbose, dynamic).await;
-            if let Some(spinner) = spinner {
-                if result.is_ok() {
-                    spinner.finish();
-                } else {
-                    spinner.failed();
+            match coffee.install(&plugin, verbose, dynamic).await {
+                Ok(_) => {
+                    spinner.and_then(|splinner| Some(splinner.finish()));
+                    term::success!("Plugin {plugin} Compiled and Installed")
+                }
+                Err(err) => {
+                    spinner.and_then(|spinner| Some(spinner.failed()));
+                    return Err(err);
                 }
-            } else if result.is_ok() {
-                term::success!("Plugin {plugin} Compiled and Installed")
             }
         }
         CoffeeCommand::Remove { plugin } => {
```

Sorry about this but I do not like too much branching in Rust for readability reasons

If clippy has something to say, we silence it :) and we make it as a repository rule; Everyone need to have a convention stile agreement

Reply:
Ah got it. One thing tho this code ignores the verbose variable and also does a `term::success` even if `verbose=false´:
```rust
            match coffee.install(&plugin, verbose, dynamic).await {
                Ok(_) => {
                    spinner.and_then(|spinner| Some(spinner.failed()));
                    if verbose {
                        term::success!("Plugin {plugin} Compiled and Installed")
                    }
                }
                Err(err) => {
                    spinner.and_then(|spinner| Some(spinner.failed()));
                    term::error(format!("{err}"))
                }
            }
```